### PR TITLE
Data processing

### PR DESCRIPTION
**Pull request by SaraKingGH**
This is the final data processing for Rogers Project.

### CODE REVIEW

File: dataprocessing/data_utils.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+from pathlib import Path
+logging.basicConfig(level=logging.INFO, format='%(levelname)s: %(message)s')
+logger = logging.getLogger(__name__)
+
+
+def videos2frames_frame_number(inputdir: Path, outputdir: Path, frame_number: int):
+    """
+    convert video to frames based on the requested number of frames.
+    """
+    assert inputdir.is_dir()
```
Comment (gkericks) [inline dataprocessing/data_utils.py]: You could just throw an exception, that might be more appropriate than an assert. Or let the exception just get thrown when the variable is used incorrectly.

File: dataprocessing/data_utils.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        logger.info(f'frame count:, {fcnt}')
+        logger.info(f'video duration:, {fcnt/fps}')
+        count = 0
+        save_count = 0
+        while True:
+            success, image = vidcap.read()
+            if not success:
+                break
+            if count % frame_number == 0:
+                cv2.imwrite(str(current_output / "frame{:d}.jpg".format(count)), image)
```
Comment (gkericks) [inline dataprocessing/data_utils.py]:
```suggestion
                cv2.imwrite(str(current_output / 'frame{:d}.jpg'.format(count)), image)
```

File: dataprocessing/data_utils.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                save_count += 1
+                logger.info(f'frame {count} is saved!')
+            count += 1
+
+
+def frame2video_convertor(pathIn: Path, pathOut: Path, fps: int):
```
Comment (gkericks) [inline dataprocessing/data_utils.py]: very minor, use snake_case for variable/parameter names

File: dataprocessing/video2frame.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+logger = logging.getLogger(__name__)
+import sys
+sys.path.append('../')
+sys.argv=['']; del sys
+from data.config import DATA_PATH
+from data_utils import videos2frames_frame_number
+
+
+if __name__ == "__main__":
+    logger.info(f"breaking the video to frames and save the results")
```
Comment (gkericks) [inline dataprocessing/video2frame.py:14]: Minor, use single quotes

File: dataprocessing/frame2video.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+sys.argv=['']; del sys
+from data.config import DATA_PATH
+from data_utils import frame2video_convertor
+
+
+if __name__ == "__main__":
+    logger.info(f"combining the frames to video and save the results")
+    a = argparse.ArgumentParser()
+    a.add_argument("--pathIn", help="path to the image folder")
+    a.add_argument("--pathOut", help="folder for saving the output vide.")
```
Comment (gkericks) [inline dataprocessing/frame2video.py]:
```suggestion
    a.add_argument('--pathOut', help='folder for saving the output video')
```

**Comment by gkericks** [comment]
Looks good! Just minor things

File: dataprocessing/data_utils.py (hunk 3 above)
Comment (SaraKingGH) [reply dataprocessing/data_utils.py]: done here 
a07481e

File: dataprocessing/video2frame.py (hunk 4 above)
Comment (SaraKingGH) [reply dataprocessing/video2frame.py:14]: all the double quotes problems solved.

File: dataprocessing/data_utils.py (hunk 1 above)
Comment (SaraKingGH) [reply dataprocessing/data_utils.py]: fixed here aed17ab

**Comment by SaraKingGH** [comment]
> Looks good! Just minor things

Thanks Graham for reviewing!

**Pull request merged**